applications/plugins/SofaPython/Binding_GUIManager.cpp

```diff
@@ -0,0 +1,249 @@
+:type path: str
+)DOC";
+static PyObject * GUIManager_setScreenshotDirectoryPath(PyObject * /*self*/, PyObject * args) {
+    const char * path_ptr;
+    PyObject * createIfNecessaryObj = nullptr;
+
+    if (!PyArg_ParseTuple(args, "s|O", &path_ptr, &createIfNecessaryObj)) {
+        SP_MESSAGE_ERROR("GUIManager.setScreenshotDirectoryPath requires a path as first argument.");
+        return nullptr;
+    }
+
+    bool createIfNecessary = (createIfNecessaryObj == nullptr || PyObject_IsTrue(createIfNecessaryObj));
+    sofa::gui::BaseGUI::setScreenshotDirectoryPath(path_ptr, createIfNecessary);
+
+    Py_RETURN_NONE;
+}
+
+static constexpr const char* getScreenshotDirectoryPath_DOC =
+R"DOC(
+Get the current Sofa's screenshot directory path.
+
+:return: The current Sofa's screenshot directory path.
+:rtype: str
+)DOC";
+static PyObject * GUIManager_getScreenshotDirectoryPath(PyObject * /*self*/, PyObject * /*args*/) {
+    const std::string & path = sofa::gui::BaseGUI::getScreenshotDirectoryPath();
+    return PyString_FromString(path.c_str());
+}
+
+static constexpr const char* setSofaPrefix_DOC =
+R"DOC(
+Set the Sofa's prefix directory path.
+
+:param path: The path to the Sofa's prefix directory
+:type path: str
+)DOC";
+static PyObject * GUIManager_setSofaPrefix(PyObject * /*self*/, PyObject * args) {
+    const char * path_ptr;
+
+    if (!PyArg_ParseTuple(args, "s", &path_ptr)) {
+        SP_MESSAGE_ERROR("GUIManager.setSofaPrefix requires a path as first argument.");
+        return nullptr;
+    }
+
+    sofa::gui::GUIManager::SetSofaPrefix(path_ptr);
+    Py_RETURN_NONE;
+}
+
+static constexpr const char* getSofaPrefix_DOC =
+R"DOC(
+Get the current Sofa's prefix directory path.
+
+:return: The current Sofa's prefix directory path.
+:rtype: str
+)DOC";
+static PyObject * GUIManager_getSofaPrefix(PyObject * /*self*/, PyObject * /*args*/) {
+    const std::string & path = sofa::gui::GUIManager::GetSofaPrefix();
+    return PyString_FromString(path.c_str());
+}
+
+static constexpr const char* init_DOC =
+R"DOC(
+Initialize the GUI manager.
+
+:param type: The GUI's type.
+:type path: str
+)DOC";
+static PyObject * GUIManager_init(PyObject * /*self*/, PyObject * args) {
+    const char * name_ptr;
+    if (!PyArg_ParseTuple(args, "s", &name_ptr)) {
+        std::string msg = "GUIManager.init requires a valid gui name. Registered GUIs are '" + sofa::gui::GUIManager::ListSupportedGUI(',') + "'";
+        SP_MESSAGE_ERROR(msg.c_str());
+        return nullptr;
+    }
+    sofa::gui::GUIManager::Init("sofa", name_ptr);
+    Py_RETURN_NONE;
+}
+
+static constexpr const char* createGUI_DOC =
+R"DOC(
+Create a new GUI.
+
+:return: 1 if the GUI creation failed, 0 otherwise
+:rtype: int
+)DOC";
+static PyObject * GUIManager_createGUI(PyObject * /*self*/, PyObject * /*args*/) {
+    int err = sofa::gui::GUIManager::createGUI();
+    return PyInt_FromLong(err);
+}
+
+static constexpr const char* setDimension_DOC =
```

Comment:
I like a lot how you make the docstring...much better than how it is done now.
Thank for the source of inspiration.